plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -606,8 +607,7 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         DiscretePredicates discretePredicates = null;
         if (!partitionSourceIds.isEmpty()) {
             // Extract identity partition columns
-            Map<Integer, IcebergColumnHandle> columns = getColumns(icebergTable.schema(), typeManager).stream()
-                    .filter(column -> partitionSourceIds.contains(column.getId()))
+            Map<Integer, IcebergColumnHandle> columns = getPartitionColumns(icebergTable.spec(), typeManager).stream()
```

Comment:
Could you update a product test? Probably, `io.trino.tests.product.iceberg.TestIcebergSparkCompatibility#testPartitionedByNestedField`

Reply:
ok, I wiil update it later